knimin/tests/test_ag_edit_barcode.py

```diff
@@ -0,0 +1,108 @@
+from unittest import main
+
+from tornado.escape import url_escape
+
+from knimin.tests.tornado_test_base import TestHandlerBase
+from knimin import db
+
+
+class TestAGEditBarcodeHandler(TestHandlerBase):
+    def test_get_not_authed(self):
+        response = self.get('/ag_edit_barcode/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/login/?next=%s' %
+                         (port, url_escape('/ag_edit_barcode/')))
+
+    def test_get(self):
+        self.mock_login_admin()
+
+        # check that error is raised for unknown barcode
+        response = self.get('/ag_edit_barcode/', {'barcode': 'unknown'})
+        self.assertEqual(response.code, 500)
+        self.assertRaises(KeyError)
```

Comment:
Does this actually work as expected? If the call to `self.get` raises a `KeyError`, the exception will happen at that point in the stack. Generally, the form used with `assertRaises` is:

```python
with self.assertRaises(SomeSpecificException):
    some_block_of_code_which_excepts
```

The context manager than catches the exception and tests that it is the expected exception